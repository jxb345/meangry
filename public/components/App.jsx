import React, { useState } from 'react';
import Signup from './Signup.jsx';
import EmailForm from './EmailForm.jsx';

function App ()  {
  const [signup, setSignup] = useState(false);
  const [emailForm, setEmailForm] = useState(false);

  return (
    <div className="grid">
          <div className="one">
            <img src="./heatMailLogo.png" height="40" width="40" />
            <p className="vertical-center">
              heatMail
            </p>

          </div>
          <div className="two">
          <br/>
          <span className="furious">
            You are FURIOUS!
          </span>
          <p>
            That email from your colleague, friend, family member, etc. sets off an emotional surge <br/> that <i>needs</i> an outlet,
            but your primitive fight response will only lead to regrets later... <br/> instead, say what you MEAN and send it off <u>anonymously</u> to a stranger.
          </p>
          </div>
          <div className="three">
              <Signup />

          </div>
          <div className="four">
              <EmailForm />

          </div>
        </div>
    )
}

export default App;