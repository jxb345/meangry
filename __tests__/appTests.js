import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import App from '../public/components/App.jsx';

test('app-snapshot-test', () => {
  const component = renderer.create(
    <App/>);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    test('preview button', () => {

      const app = shallow(<App />);
      expect(app.find('button').text()).toEqual('Preview');
  })
})
