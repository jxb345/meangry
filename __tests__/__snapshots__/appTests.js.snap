// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app-snapshot-test 1`] = `
<div
  className="grid"
>
  <div
    className="grid-title"
  >
    <img
      height="40"
      src="./heatMailLogo.png"
      width="40"
    />
    <p
      className="vertical-center"
    >
      heatMail
    </p>
  </div>
  <div
    className="grid-text"
  >
    <div
      className="explanation-text-wrapper"
    >
      <div
        className="you-are-title"
      >
        You feel 
        <img
          alt="thought bubble"
          className="thought-bubble-img"
          height="80"
          src="./bubble.png"
          width="110"
        />
      </div>
      <div
        className="explanation-text"
      >
        That message ignites an emotional surge  
        <br />
         that 
        <i>
          needs
        </i>
         an outlet - but DON'T 
        <span>
           
          <a
            href="http://localhost:3600/doNotReply"
          >
            <button
              className="reply-graphic"
              title="do NOT reply"
            >
              REPLY
            </button>
          </a>
        </span>
        <span
          className="explanation-text-last"
        >
          <br />
           Instead, send an 
          <strong>
            anon
          </strong>
           note to a random.
        </span>
      </div>
    </div>
  </div>
  <div
    className="grid-signup"
  >
    <div>
      <p
        className="receive-text"
      >
        ...
        <i>
          receive
        </i>
         anonymous email:
      </p>
      <form>
        <input
          className="email-signup"
          onChange={[Function]}
          placeholder="your email address"
          required={true}
          type="text"
        />
        <br />
        <button
          className="signup-button"
          id="signup"
          onClick={[Function]}
          type="button"
        >
          Sign Up
        </button>
      </form>
      <div>
        <br />
        <br />
      </div>
      <div
        className="signup-tally"
      >
        0
         signed up to receive heatMail.
      </div>
    </div>
  </div>
  <div
    className="grid-body"
  >
    <div>
      <input
        className="subject"
        id="subject"
        onChange={[Function]}
        placeholder="I feel _______"
        type="text"
      />
      <br />
      <textarea
        className="body"
        cols="69"
        id="body"
        onChange={[Function]}
        placeholder="because..."
        rows="11"
        type="text"
      />
      <br />
    </div>
  </div>
  <div
    className="grid-buttons"
  >
    <button
      className="button-preview"
      onClick={[Function]}
      type="button"
    >
      Preview
    </button>
    <div
      className="preview-button"
    >
      <input
        defaultChecked={true}
        id="preview"
        name="preview"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="preview"
      >
          preview before sending email
      </label>
    </div>
  </div>
  <div />
</div>
`;
